File: A2/A2-003/A2-003.py
# Input trees height
n = int(input())
trees = list(map(int, input().split()))

# Count amount of trees that has no taller adjacent trees
ans = 0
if(trees[0] >= trees[1]):
    ans += 1
for i in range(1, n - 1):
    if(trees[i] >= trees[i - 1] and trees[i] >= trees[i + 1]):
        ans += 1
if(trees[n - 1] >= trees[n - 2]):
    ans += 1

# Output
print(ans)

File: A2/A2-003/A2-003.cpp
#include<iostream>
#include<vector>
using namespace std;

int n, ans = 0;
vector<int> trees;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);
    
    // Input trees height
    cin >> n;
    trees.resize(n);
    for(int i = 0; i < n; i++) { cin >> trees[i]; }
    // Count amount of trees that has no taller adjacent trees
    if(trees[0] >= trees[1]) { ans++; }
    for(int i = 1; i < n - 1; i++)
    {
        if(trees[i] >= trees[i - 1] && trees[i] >= trees[i + 1])
        {
            ans++;
        }
    }
    if(trees[n - 1] >= trees[n - 2]) { ans++; }
    // Output
    cout << ans;
    return 0;
}